cli-plugins/socket/socket.go

```diff
@@ -17,7 +17,7 @@ const EnvKey = "DOCKER_CLI_PLUGIN_SOCKET"
 // and update the conn pointer, and returns the listener for the socket (which the caller
 // is responsible for closing when it's no longer needed).
 func SetupConn(conn **net.UnixConn) (*net.UnixListener, error) {
-	listener, err := listen("docker_cli_" + uuid.Generate().String())
+	listener, err := listen("docker_cli_" + uuid.New().String())
```

Comment:
Or at least, yeah, I _think_ it should be random enough (backed by `crypto/rand`), and it would not introduce a new _direct_ dependency. (google/uuid is not _too problematic_ to have as dependency, but yeah, less is always good!)

Let me work on that; I can push it as a new commit, and then I can squash

Reply:
Sounds good to me as well, i doubt we'll ever have enough plugin sockets open to even consider the "actual randomness" of this solution (and, then again, it's based on `crypto/rand` so i see no real issues)

Reply:
Some discussion whether we need the extra functionality of `GenerateRandomID` to check if it might be a number. I just did a quick benchmark what it would be like if we did not have that (as it's not needed for this purpose);

```bash
goos: darwin
goarch: arm64
pkg: github.com/docker/docker/pkg/stringid
BenchmarkGenerateRandomID
BenchmarkGenerateRandomID-10           	 1899510	       632.7 ns/op	     240 B/op	       6 allocs/op
BenchmarkGenerateRandomIDNoCheck
BenchmarkGenerateRandomIDNoCheck-10    	 2148127	       563.1 ns/op	     160 B/op	       3 allocs/op
```

As it's only 5 lines of code, I'd also be good to  define a local utility instead;

```go
func GenerateRandomID() string {
	b := make([]byte, 32)
	if _, err := rand.Read(b); err != nil {
		panic(err) // This shouldn't happen
	}
	return hex.EncodeToString(b)
}
```